static/skin/index.js

```diff
@@ -279,26 +279,31 @@
         });
     }
 
-    function noResultsContent(kiwixHomeBody) {
+    function resetAllFiltersAndRedirect() {
+        window.history.pushState({}, null, `${window.location.href.split('?')[0]}?lang=`);
```

Comment:
1. Under Firefox this can be shortened to a clearer `window.history.pushState({}, null, '?lang=');`. If other browsers handle relative urls in `window.history.pushState()` equally well, then it makes sense to simplify all usages of that method in this file.
2. Also it makes sense to introduce a small helper function `pushFilterState()` that combines `window.history.pushState()` with `setCookie(filterCookieName, ...)`.